src/include/duckdb/catalog/catalog_entry/index_catalog_entry.hpp

```diff
@@ -24,21 +22,37 @@ class IndexCatalogEntry : public StandardEntry {
 	static constexpr const char *Name = "index";
 
 public:
-	//! Create an IndexCatalogEntry and initialize storage for it
+	//! Create an IndexCatalogEntry
 	IndexCatalogEntry(Catalog &catalog, SchemaCatalogEntry &schema, CreateIndexInfo &info);
 
-	optional_ptr<Index> index;
+	//! The SQL of the CREATE INDEX statement
 	string sql;
+	//! Additional index options
+	case_insensitive_map_t<Value> options;
+
+	//! The index type (ART, B+-tree, Skip-List, ...)
+	IndexType index_type;
```

Comment:
Could you move this to the `DuckIndexEntry` instead?

Reply:
We also need these fields for (de)serialization in `CreateIndexInfo`. And on second thought, I think these fields apply to many indexes, not specifically the `DuckIndexEntry`, no? Or would I create a separate `CreateDuckIndexInfo`? I didn't see that for any other entry, e.g., there is no `CreateDuckTableInfo`.